### ISSUE TITLE
Collect process ID to detect app recycle/crash

### ISSUE THREAD

**Issue by cijothomas** [Microsoft/ApplicationInsights-dotnet-server#620]
As mentioned here, https://github.com/Microsoft/ApplicationInsights-dotnet-server/issues/340, AppInsights don't detect crashes or app recycles. If we simply collect the process id with each telemetry, then analytics query can be written to figure out how often process has recycled by looking at change in process ids.

**Comment by cijothomas** [comment]
Processid is already read in perf counters module to find the instancename for reading perf counter.
https://github.com/Microsoft/ApplicationInsights-dotnet-server/blob/2a681f5399a6aaee554fa2d93a1d6447a8402fe1/Src/PerformanceCollector/Shared/Implementation/PerformanceCounterUtility.cs#L265

**Comment by SergeyKanzhelev** [comment]
"simply collect the process id with each telemetry" - this is dead end. We should not collect process ID with every telemetry item - it is quite expensive. Think of heartbeat/discovery events or performance counter instead.

**Comment by pharring** [comment]
Also, for IIS the process ID (PID) will be that of w3wp.exe. It won't change for app recycles. It changes only if IIS itself is restarted (or crashes).

**Comment by cijothomas** [comment]
@pharring Thanks for correcting. We can still detect how often app has crashed by looking at pids - right?

(btw from what i read, i thought app recycling cause change to pid. https://www.outsystems.com/forums/discussion/10307/understanding-application-pool-recycling-and-how-to-fine-tune-for-use-with-outsys/)

**Comment by SergeyKanzhelev** [comment]
Expensive in terms of dollars for customers to pay

**Comment by pharring** [comment]
@cijothomas Thanks for that link. I've observed application restarts in Azure App Service where the w3wp process does not get restarted. I admit that I jumped to the conclusion that it was App Pool recycling, but that article indicates that it couldn't be that. I don't know what the behavior I've observed is really called. 

Oh... AppDomains. Remember to record the AppDomain ID as well as the PID.

Note that Windows re-uses PIDs too, but I think you'd be incredibly unlucky to get the same PID twice (given that there's probably some overlap between one worker process going down and another spinning up)
